### PR TITLE
feat(installer): add ingress to control-plane helm chart keptn installer (#5066)

### LINKED ISSUE #5066: Add Ingress option to Keptn installer manifests for control plane
**Is your feature request related to a problem? Please describe.**
As I would like to use the GitOps paradigm for deploying Keptn to my Kubernetes Cluster
Then I would like the ability to specify ingress in the values.yaml
When Deploying with ArgoCD so I don't have to run ingress separate after

I would much rather it be an option in the helm chart so I have a clean deployment, and everything is declared in one place, instead of running it after.

**Describe the solution you'd like**
This ingress.yaml should be in the installer/manifests/keptn/charts/control-plane/templates and its config in the values.yaml under a new section called ingress (if under apiGatewayNginx it could create confusion).

The solution needs to have the ability to say enabled: true/false this will not break any existing or future deployments or config.

It also needs to cater for annotations, hosts, and tls as options

**Describe alternatives you've considered**
Alternative solutions are to not have this and just create my ingress after. eg kubectl create -f ingress.yaml

**Additional context**
I'm happy to write this code myself

### PR DESCRIPTION
**Pull request by bradmccoydev**
Signed-off-by: bradmccoydev <bradmccoydev@gmail.com>

## This PR
This adds the ability to add your ingress for the keptn helm chart installer so you don't have to do it after eg installing with ArgoCD

### Related Issues
Resolves Issue: https://github.com/keptn/keptn/issues/5066

Fixes #5066 

### How to test
helm install control-plane --generate-name --dry-run --debug
```
# Test enabled : false
ingress:
  enabled: false
  annotations: {}
  host:
    []
  path: /
  pathType: Prefix
  tls:
    []
---
null
---

# Test One
ingress:
  enabled: true
  annotations: {}
  host:
    []
  path: /
  pathType: Prefix
  tls:
    []
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: ingress
  labels:
    app.kubernetes.io/name: ingress
    app.kubernetes.io/instance: control-plane-1631932093
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/part-of: keptn-default
    app.kubernetes.io/component: control-plane
    helm.sh/chart: control-plane-0.1.0
spec:
  rules:
    - http:
        paths:
          - path: /
            pathType: Prefix
            backend:
              service:
                name: api-gateway-nginx
                port:
                  number: 80
---
# Test Two
ingress:
  enabled: true
  annotations:
    kubernetes.io/ingress.class: nginx
    kubernetes.io/tls-acme: "true"
  host:
    []
  path: /
  pathType: Prefix
  tls:
    []
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: "nginx"
    kubernetes.io/tls-acme: "true"
  name: ingress
  labels:
    app.kubernetes.io/name: ingress
    app.kubernetes.io/instance: control-plane-1631932452
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/part-of: keptn-default
    app.kubernetes.io/component: control-plane
    helm.sh/chart: control-plane-0.1.0
spec:
  rules:
    - http:
        paths:
          - path: /
            pathType: Prefix
            backend:
              service:
                name: api-gateway-nginx
                port:
                  number: 80
---
# Test 3
ingress:
  enabled: true
  annotations:
    kubernetes.io/ingress.class: nginx
  host: "keptn.bradmccoy.io"
  path: /
  pathType: Prefix
  tls:
    []
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: "nginx"
  name: keptn-ingress
  namespace: keptn
  labels:
    app.kubernetes.io/name: ingress
    app.kubernetes.io/instance: control-plane-1631932981
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/part-of: keptn-default
    app.kubernetes.io/component: control-plane
    helm.sh/chart: control-plane-0.1.0
spec:
  rules:
    - host: keptn.bradmccoy.io
      http:
        paths:
          - path: /
            pathType: Prefix
            backend:
              service:
                name: api-gateway-nginx
                port:
                  number: 80
---
# Test 4
ingress:
  enabled: true
  annotations:
    kubernetes.io/ingress.class: nginx
    kubernetes.io/tls-acme: "true"
  host: "keptn.bradmccoy.io"
  path: /
  pathType: Prefix
  tls:
    - secretName: keptn-tls-certificate
      hosts:
        - keptn.bradmccoy.io
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: "nginx"
    kubernetes.io/tls-acme: "true"
  name: keptn-ingress
  namespace: keptn
  labels:
    app.kubernetes.io/name: ingress
    app.kubernetes.io/instance: control-plane-1631933890
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/part-of: keptn-default
    app.kubernetes.io/component: control-plane
    helm.sh/chart: control-plane-0.1.0
spec:
  rules:
    - host: keptn.bradmccoy.io
      http:
        paths:
          - path: /
            pathType: Prefix
            backend:
              service:
                name: api-gateway-nginx
                port:
                  number: 80
  tls:
    - hosts:
      - keptn.bradmccoy.io
      secretName: keptn-tls-certificate
---
```

### CODE REVIEW

**Comment by mowies** [comment]
Hi @bradmccoydev !
Thanks for the contribution, this makes a lot of sense in my opinion.
I created a follow-up issue to add this new option to our documentation as well.

File: installer/manifests/keptn/charts/control-plane/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+              service:
+                name: api-gateway-nginx
+                port:
+                  number: {{ .Values.apiGatewayNginx.port }}
+  {{- end -}}
+  {{- if .Values.ingress.tls }}
+  tls:
+    {{- toYaml .Values.ingress.tls | nindent 4 }}
+  {{- end -}}
+{{- end -}}
```
Comment (mowies) [inline installer/manifests/keptn/charts/control-plane/templates/ingress.yaml:48]: Please add the missing newline :)

File: installer/manifests/keptn/charts/control-plane/values.yaml [hunk 2]
```diff
@@ -129,3 +129,12 @@ alpine:
...
     tag: '3.13'
+
+ingress:
+  enabled: false
+  annotations: {}
+  host: {}
+  path: /
+  pathType: Prefix
+  tls:
+    []
```
Comment (mowies) [inline installer/manifests/keptn/charts/control-plane/values.yaml:140]: Please add the missing newline :)

File: installer/manifests/keptn/charts/control-plane/templates/ingress.yaml (hunk 1 above)
Comment (bradmccoydev) [reply installer/manifests/keptn/charts/control-plane/templates/ingress.yaml]: Hi @mowies can you please check now

File: installer/manifests/keptn/charts/control-plane/values.yaml (hunk 2 above)
Comment (bradmccoydev) [reply installer/manifests/keptn/charts/control-plane/values.yaml]: Hi @mowies can you please check now

**Comment by mowies** [comment]
What happened here with the merge? 
seems like github picked up some merged changes as new changes 😅

**Comment by mowies** [comment]
@bradmccoydev there seems to be a commit that doesn't have the signoff annotation. could you please double-check? this should fix the DCO PR check.
you can find more info under the [failing PR check](https://github.com/keptn/keptn/pull/5293/checks?check_run_id=3649226100)

**Comment by bradmccoydev** [comment]
Hey @mowies fixing now, cheers

**Comment by bradmccoydev** [comment]
@mowies looks like that worked with the signoff

**Comment by mowies** [comment]
signoff is correct now but there's still conflicts that came from merge artifacts it seems. there's 4 commits that should not be in this branch at all but they are still here for some reason. (all the fix:... commits)

**Comment by bradmccoydev** [comment]
Closing PR due to rabbit hole

**Comment by mowies** [comment]
Should be easy to cherry-pick your 2 commits into a new branch and then open a new PR

**Pull request closed without merging**